### ISSUE TITLE
supporting attributes for GraphML formats

### ISSUE THREAD

**Issue by filchristou** [JuliaGraphs/GraphIO.jl#47]
I would like to be able to read a metagraph from a graphML format.
This means reading in all attributes and properties that a graphML file may contain, and storing them in a metagraph.
I 've been playing around for the last hours with the package and I got it going.

Obviously the packet dependencies will grow (MetaGraphs)

The implementation is done using XPath and DOM and not the `EzXML.StreamReader` for the reason that there needs to be some memory on the parser in order to remember whose are the properties read.
I don't argue that it is impossible to do with `EzXML.StreamReader` and I also understand that reading all structure in RAM is more resource-hungry. 
I just implemented  the obvious solution

Is it fine if I do a PR ?

**Comment by filchristou** [comment]
Yeah, I put the functionality there because stuff was going slow here. I don't want to register NestedGraphsIO.jl, because ideally it should be an extension to GraphIO.jl with the `NestedGraphs` weak dependency. Moreover, I am having some second thoughts with `NestedGraphs.jl` because I know find it a bit overengineered. Ideally, I would like to reimplement it using subgraphs or [views](https://github.com/JuliaGraphs/GraphsBase.jl/issues/8) so that they are better integrated into the ecosystem. But I really lack the time now and Graphs.jl is not there yet.

**Comment by Krastanov** [comment]
Actually, there seems to be an even quicker solution, also by @filchristou . Check out this stagnated PR https://github.com/JuliaGraphs/GraphIO.jl/pull/55 -- is this a sufficient solution?
